plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/expression/MongoExpression.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.mongodb.expression;
+
+import org.bson.Document;
+
+import java.util.Optional;
+
+public record MongoExpression(Object expression)
+{
+    public Optional<Document> expressionDocument()
+    {
+        if (expression instanceof Document document) {
+            return Optional.of(document);
+        }
+        return Optional.empty();
+    }
```

Comment:
What is the reason behind this wrapping ?

Reply:
ReWriter can return any type (Integer, String, Document ..) of the object, So I wanted to wrap all of these in this class.